### PR TITLE
Upgrade laminas/laminas-diactoros

### PR DESCRIPTION
**Pull request by s0lus**
For support php 8.3

### CODE REVIEW

**Comment by s0lus** [comment]
@mcustiel, hey! Looks like Scrutinizer doesn't support php 8.3. Do you mind if I switch to github actions?

**Comment by s0lus** [comment]
@mcustiel ping

**Comment by s0lus** [comment]
@mcustiel ping

**Comment by mcustiel** [comment]
Hello @s0lus 
Sorry for the delay, I was away from computers during the last month. As soon as I have enough time I will have a look at your PR. In advance: thank you very much for taking time to do changes and improvements.

File: composer.json [hunk 1]
```diff
@@ -25,8 +25,8 @@
     "description" : "Common libraries for Phiremock",
     "license" : "GPL-3.0-or-later",
     "require" : {
-        "php" : "^7.2|^8.0",
-        "laminas/laminas-diactoros": "^2.0",
+        "php" : "^8.1",
```
Comment (mcustiel) [inline composer.json]: I think this change is not needed, the previous version condition already accepted php8.1.

File: composer.json [hunk 2]
```diff
@@ -25,8 +25,8 @@
...
+        "laminas/laminas-diactoros": "^3.3",
```
Comment (mcustiel) [inline composer.json]: Could you write this as "^2.0|^3.0"? That way the old version is still supported for people using it.
If we change major versions of dependencies, that would need to release a major version of this library, which will lead to release a major version of phiremock-server and phiremock-client, which will lead to release a major version of the codeception module and extension. And I think we can avoid it.

File: .scrutinizer.yml [hunk 3]
```diff
@@ -19,31 +19,27 @@ build:
           file: build/coverage.xml
           format: clover
   nodes:
-    php72:
```
Comment (mcustiel) [inline .scrutinizer.yml:22]: These should remain if we keep php7.2 as a supported version.

**Comment by mcustiel** [comment]
> @mcustiel, hey! Looks like Scrutinizer doesn't support php 8.3. Do you mind if I switch to github actions?

Hello @s0lus, for me that would not be a problem as long as it runs all the nice quality checks that scrutinizer runs. Thank you for the proposal :)

File: composer.json (hunk 1 above)
Comment (s0lus) [reply composer.json]: Fixed

File: .scrutinizer.yml (hunk 3 above)
Comment (s0lus) [reply .scrutinizer.yml:22]: Fixed

File: composer.json (hunk 2 above)
Comment (s0lus) [reply composer.json]: Fixed

**Comment by s0lus** [comment]
@mcustiel Hey! I think it's ready for review, but there is a problem with coverage badge and code quality score from Scrutinizer.

I didn't find any alternative for code quality score badge, but for code coverage we can use [codecov.](https://about.codecov.io/tool/phpunit/) I don't have permission to set up codecov in your project, so it's better that you to do that.

Also, github actions not displaying on your repo, but I think after merge it will. Anyway to prove that this works here pipeline from my fork https://github.com/s0lus/phiremock-common/actions

**Comment by s0lus** [comment]
@mcustiel friendly ping :)

**Review by mcustiel — APPROVED** [review]

**Comment by mcustiel** [comment]
Thank you very much for your help @s0lus

**Pull request merged**